### ISSUE TITLE
Ruby 2.3.0 Timeout Deprecation Warnings

### ISSUE THREAD

**Issue by hcatlin** [nahi/httpclient#289]
It seems that in Ruby 2.3.0, the Object#timeout method is fully deprecated, and was only a helper method aliasing Timeout#timeout.

It seems that replacing `timeout` with `Timeout.timeout` should work for all versions that you might be concerned with.

A quick way to solve this would be to include the module into the session.rb file.

Currently, my application gives these: 

````
/Users/hcatlin/.rvm/gems/ruby-2.3.0/gems/httpclient-2.7.0.1/lib/httpclient/session.rb:787:in `parse_header': Object#timeout is deprecated, use Timeout.timeout instead.
/Users/hcatlin/.rvm/gems/ruby-2.3.0/gems/httpclient-2.7.0.1/lib/httpclient/session.rb:867:in `read_body_length': Object#timeout is deprecated, use Timeout.timeout instead.
/Users/hcatlin/.rvm/gems/ruby-2.3.0/gems/httpclient-2.7.0.1/lib/httpclient/session.rb:500:in `query': Object#timeout is deprecated, use Timeout.timeout instead.
/Users/hcatlin/.rvm/gems/ruby-2.3.0/gems/httpclient-2.7.0.1/lib/httpclient/session.rb:787:in `parse_header': Object#timeout is deprecated, use Timeout.timeout instead.
/Users/hcatlin/.rvm/gems/ruby-2.3.0/gems/httpclient-2.7.0.1/lib/httpclient/session.rb:867:in `read_body_length': Object#timeout is deprecated, use Timeout.timeout instead.
/Users/hcatlin/.rvm/gems/ruby-2.3.0/gems/httpclient-2.7.0.1/lib/httpclient/session.rb:500:in `query': Object#timeout is deprecated, use Timeout.timeout instead.
/Users/hcatlin/.rvm/gems/ruby-2.3.0/gems/httpclient-2.7.0.1/lib/httpclient/session.rb:787:in `parse_header': Object#timeout is deprecated, use Timeout.timeout instead.
/Users/hcatlin/.rvm/gems/ruby-2.3.0/gems/httpclient-2.7.0.1/lib/httpclient/session.rb:867:in `read_body_length': Object#timeout is deprecated, use Timeout.timeout instead.
````

**Comment by elifoster** [comment]
This was resolved by #287. @nahi You should close this.

**Comment by hcatlin** [comment]
Yeah, I just saw that.
